### DISCUSSION TITLE
Why use all data？

### DISCUSSION THREAD

**Discussion by Chrisable** [pyg-team/pytorch_geometric#4771] [General]
Why not use the training data instead of all the data. Is it because GCN is transductive learning rather than inductive learning？If we use graphsage model , Is training data used instead of all data？
![image](https://user-images.githubusercontent.com/49330390/172114353-7d6fce73-07cd-471e-9ff0-ddd859b7ff6c.png)

**Comment by rusty1s** [comment]
Yes, this relates to transductive learning. The GNN makes use of all available data, while supervision/loss computation is only performed on the training nodes. If you want to move this to an inductive learning scenario, you could do
```python
train_data = data.subgraph(data.train_mask)
```
and input this into the GNN instead.

**Reply by Chrisable** [reply]
when i use graphsage model without NeighborLoader,does it mean sample all neighbor ?
code: https://github.com/ytchx1999/PyG-GraphSAGE/blob/main/GraphSAGE_pyg.ipynb

**Reply by rusty1s** [reply]
Yes, the `NeighborLoader` is independent from learning in a transductive or inductive scenario. It will just sample neighbors from the graph which it gets as input, e.g., either `data.edge_index` for transductive learning, or `train_data.edge_index` for inductive learning.

**Reply by Chrisable** [reply]
Thank you for your reply.By the way,when i use the formula “train_data = data.subgraph(data.train_mask)”,i found that train_data.edge_index+val_data.edge_index+test.edge_index is not equal to data.edge_index.Is it because of the existence of edge cuts? If so,this will bring about a drop in accuracy. Is inductive learning generally handled this way?

**Reply by rusty1s** [reply]
Yes, this is correct. You will not maintain inter-edges between different splits. As such, transductive learning is generally considered the superior approach when learning on a single graph.
